src/changelog.ts

```diff
@@ -246,22 +286,22 @@ class Changelog {
    * Changes are migrated in their existing categories, and placed above any
    * pre-existing changes in that category.
    *
-   * @param {Version} version - The release version to migrate unreleased
-   *   changes to.
+   * @param version - The release version to migrate unreleased
+   * changes to.
    */
-  migrateUnreleasedChangesToRelease(version) {
+  migrateUnreleasedChangesToRelease(version: Version) {
     const releaseChanges = this._changes[version];
     if (!releaseChanges) {
       throw new Error(`Specified release version does not exist: '${version}'`);
     }
 
     const unreleasedChanges = this._changes[unreleased];
 
-    for (const category of Object.keys(unreleasedChanges)) {
-      if (releaseChanges[category]) {
+    for (const category of Object.keys(unreleasedChanges) as ChangeCategory[]) {
+      if (category in releaseChanges) {
         releaseChanges[category] = [
-          ...unreleasedChanges[category],
-          ...releaseChanges[category],
+          ...(unreleasedChanges[category] as string[]),
+          ...(releaseChanges[category] as string[]),
```

Comment:
The issue with this cast seems to be that TypeScript isn't good at inferring types of object properties.

Line 301 had been `if (releaseChanges[category])` beforehand, which seems a bit better than what's there now. But it still didn't trigger the type inference to rule out `undefined`. I was able to get it to work by pulling it out into a separate variable. e.g.
```
const releaseChanges = releaseChangeCategories[category];
if (releaseChanges) {
  releaseChanges.unshift(...(unreleasedChanges as string[]));
} else {
  releaseChangeCategories[category] = unreleasedChanges;
}
```

Which... isn't great, but demonstrates what I'm talking about at least. Oh also I renamed the variable `releaseChanges` to `releaseChangeCategories` so I could use the name `releaseChanges` for a specific category, and likewise for the unreleased changes.

Reply:
I restored `if (releaseChanges[category])`, but otherwise left things as they are. I prefer spreading over `.unshift`, and `.unshift` didn't get rid of _all_ casts anyway. 3cc4e79